components/chart/linearTimeSeries.js

```diff
@@ -0,0 +1,127 @@
+                    .attr('class', 'background')
+                    .layout({
+                        position: 'absolute',
+                        top: 0,
+                        right: 0,
+                        bottom: 0,
+                        left: 0
+                    });
+
+                var gridlinesContainer = mainContainer.selectAll('g.gridlines')
+                    .data([data]);
+                gridlinesContainer.enter()
+                    .append('g')
+                    .attr('class', 'gridlines');
+
+                var yAxisContainer = mainContainer.selectAll('g.y-axis')
+                    .data([data]);
+                yAxisContainer.enter()
+                    .append('g')
+                    .attr('class', 'axis y-axis')
+                    .layout({
+                        position: 'absolute',
+                        top: 0,
+                        right: 0,
+                        bottom: 0
+                    });
+
+                var seriesContainer = mainContainer.selectAll('g.series')
+                    .data([data]);
+                seriesContainer.enter()
+                    .append('g')
+                    .attr('class', 'series')
+                    .layout({
+                        position: 'absolute',
+                        top: 0,
+                        right: 0,
+                        bottom: 0,
+                        left: 0
+                    });
+
+                var xAxisContainer = container.selectAll('g.x-axis')
+                    .data([data]);
+                xAxisContainer.enter()
+                    .append('g')
+                    .attr('class', 'axis x-axis')
+                    .layout('height', 20);
+
+                linearTimeSeries.layout.value(container);
+
+                xScale.range([0, xAxisContainer.layout('width')]);
+
+                yScale.range([yAxisContainer.layout('height'), 0]);
+
+                xAxisContainer.call(xAxis);
+
+                yAxisContainer.call(yAxis);
+
+                var gridlines = linearTimeSeries.gridlines.value;
+                gridlines.xScale(xScale)
+                    .yScale(yScale);
+                gridlinesContainer.call(gridlines);
+
+                var series = linearTimeSeries.series.value;
+                series.xScale(xScale)
+                    .yScale(yScale);
+                seriesContainer.call(series);
+
+            });
+        };
+
+        fc.utilities.rebind(linearTimeSeries, xScale, {
+            xDomain: 'domain',
+            xNice: 'nice'
+        });
+
+        fc.utilities.rebind(linearTimeSeries, yScale, {
+            yDomain: 'domain',
+            yNice: 'nice'
+        });
+
+        fc.utilities.rebind(linearTimeSeries, xAxis, {
+            xTicks: 'ticks'
+        });
+
+        fc.utilities.rebind(linearTimeSeries, yAxis, {
+            yTicks: 'ticks'
+        });
+
+        linearTimeSeries.layout = fc.utilities.property(fc.utilities.layout());
```

Comment:
I think we have a problem when it comes to the composability of layouts i.e. a layout of charts which themselves use layout internally is going to do a lot of laying out! 
However, I don't think this is the right approach either. I'll remove it for now.